tests/test_contrib_loader.py

```
@@ -579,5 +579,40 @@ def test_replace_css_re(self):
         self.assertEqual(l.get_output_value('url'), [u'scrapy.org'])
 
 
+class JsonProcessorTestCase(unittest.TestCase):
+        test_list_equals = {
+                                'simple': ('foo.bar', '{"foo": {"bar": "baz"}}', "baz"),
+                                'invalid': ('foo.bar.baz', '{"foo": {"bar": "baz"}}', None),
+                                'top_level': ('foo', '{"foo": {"bar": "baz"}}', {"bar": "baz"}),
+                                'double_vs_single_quoteString': ('foo.bar', "{'foo':{'bar':'baz'}}", 'baz'),
+                                'empty_string': ('foo', "", None),
+                             }
+        test_error = {
+                        'basic': ('foo.bar', '{"foo":{"bar":"baz"}}')
+                      }
+
+        def test_equals(self):
+            try:
+                import jmespath
+                for l in self.test_list_equals:
+                    expr, test_list, expected = self.test_list_equals[l]
+                    test = JsonProcessor(expr)(test_list)
+                    self.assertEqual(test, expected,
+                                     msg='test "{}" got {} expected {}'.format(l, test, expected)
+                                     )
+            except:  # in case jmespath isn't installed
```

if jmespath isn't installed just let it fail.